### DISCUSSION TITLE
Is it possible to add a custom tool?

### DISCUSSION THREAD

**Discussion by RobertsG** [awcodes/filament-tiptap-editor#158] [Q&A]
Hi there,

I'm trying to find out if its possible to add a custom tool to the toolbar, I'm in need of adding custom icons to the editor, so was wondering if its possible to specify a custom toolbar item in which I would be able to specify the html or image sources of icons that could be added in the editor content when selected.

Thanks.

**Comment by awcodes** [comment]
In v2 you can follow https://gist.github.com/awcodes/fb0425c4a2afd0fc8c3318784d9a53fb. Haven't supported it yet in v3.

**Comment by RobertsG** [comment]
Thanks for the response, interested in this in v3. Do you have any plans on implementing this in the near future?

**Reply by awcodes** [reply]
I'm working on a replacement for this plugin that will make all of this a lot easier.
